[6.5][NET] Added application service description (#1494)
Samsung/tizen-docs: docs/application/dotnet/guides/applications/uiapplication/ui-app.md

# UI Application

> Before reading this document, we recommend to familiarize with the [Application Lifecycle documentation](./application_lifecycle.md).

Reviewer: ```suggestion
Before reading this document, we recommend you get familiarized with the [Application Lifecycle documentation](./application_lifecycle.md).
```

    change events. You can override the [methods triggered for these
    events](#callback) to react to them.

- Manage [application states and transitions](./application_lifecycle.md) during the
    application life-cycle.

- Define a [background category](#allow_bg) for your application, if
    you want it to run in the background.

<a name="callback"></a>

> ## Event handling

Reviewer: ```suggestion
## Handle events
```

> ## Event handling

Reviewer: https://github.com/Samsung/tizen-docs/pull/1494#discussion_r783851602

I can't tell if this gerund case has been addressed or not.

> - Manage [application states and transitions](#state_trans) during the

Reviewer: Line 16: "Manage [application states and transitions]..." needs to be updated, the hyperlink is leading to Application States and Transitions section with anchor #state_trans, but as the section was removed, it is showing the wrong link.
There are two solutions you can incorporate:
1. Just deleting line 16~17
2. Update this link to go to application_lifecycle.md, instead of inserting line 3? This sentence wants to give a link to "Application states & transitions" anyway, which are in application_lifecycle.md now.

    application life-cycle.

- Define a [background category](#allow_bg) for your application, if
    you want it to run in the background.

<a name="callback"></a>

> ## Event Handling

Reviewer: 2nd headings should use sentence case capitalization. Also, this seems to be a case of a gerund. Perhaps a more appropriate heading would be "**_Handle events_**"
@mijin85cho can you please confirm this assessment?
```suggestion
## Event handling
```

<a name="state_trans"></a>
## Application States and Transitions

The Tizen .NET application can be in one of several application states.

The
[Tizen.Applications](/application/dotnet/api/TizenFX/latest/api/Tizen.Applications.html)
namespace defines 5 states with corresponding state change methods. A
state change method is triggered after each state change: whenever the
application is created, starts running, or is paused, resumed, or
terminated. The application must [react to each state change
appropriately](#fundamentals).

**Table: Application states**

| State        | Description                              |
|------------|----------------------------------------|
| `READY`      | Application is launched.                 |
| `CREATED`    | Application starts the main loop.        |
| `RUNNING`    | Application is running and visible to the user. |
| `PAUSED`     | Application is running but invisible to the user. |
| `TERMINATED` | Application is terminated.               |

The following figure illustrates the application state transitions.

**Figure: Application state transitions**

![Application state

> transitions](./media/app_state_transitions_cs.png)

Reviewer 2: guides/applications/uiapplication/media/app_state_transitions_cs.png was used, but with this update, now this image is not in use.

If guides/applications/uiapplication/media/app_state_transitions_cs.png will not be used any more, how about removing it?

<a name="state_trans"></a>

> ## Application States and Transitions

Reviewer: ```suggestion
## Application states and transitions
```

> ## Prerequisites

Reviewer: Line 53: I think this should be a 3rd heading under "Application states and transitions" right?

```suggestion
### Prerequisites
```

>     `Tizen.Applications` namespace. It only builds and runs.

Reviewer: Code snippet should be preceded by a colon.
```suggestion
    `Tizen.Applications` namespace. It only builds and runs:
```

</manifest>
```

## Related information

>   * Dependencies

Reviewer: Although the "*" appears as a bullet point in the staging area, requesting to use a dash "-" here for consistencies sake.
```suggestion
  - Dependencies
```

</manifest>
```

> ## Related Information

Reviewer: ```suggestion
## Related information
```
